tests/test_backend.py

```diff
@@ -429,6 +430,31 @@ def bad(run):
         assert set(f) == {".reduced", "good", "start_time"}
         assert set(f[".reduced"]) == {"good", "start_time"}
 
+def test_results_cell(mock_run, tmp_path):
+    # Test returning an object we can't save in HDF5
+    ctx_code = """
+    from damnit_ctx import Variable, Cell
+    import numpy as np
+
+    @Variable()
+    def var1(run):
+        return 7
+
+    @Variable(summary='sum')
+    def var2(run):
+        return Cell(np.arange(10), bold=False, background='#bbdddd')
+    """
+    bad_obj_ctx = mkcontext(ctx_code)
+    results = bad_obj_ctx.execute(mock_run, 1000, 123, {})
+    results_hdf5_path = tmp_path / 'results.h5'
+    results.save_hdf5(results_hdf5_path)
+    with h5py.File(results_hdf5_path) as f:
+        assert f['.reduced/var2'][()] == 45
+        assert f['.reduced/var2'].attrs['bold'] == False
+        np.testing.assert_array_equal(
+            f['.reduced/var2'].attrs['background'], [0xbb, 0xdd, 0xdd]
+        )
```

Comment:
It'd be good to have a test for a custom `summary_value` too.

Reply:
It is good, adding it immediately caught a bug. :slightly_smiling_face: